[디디] 로또 수동 입력

fucct opened the pull request:
감사합니다

fucct committed:
feat: 로또 수동

수동 로또 생성 기능 구현완료

fucct committed:
refactor : LottoCount 의존성 약화(Money 제거)

fucct committed:
refactor : 오탈자 수정

fucct committed:
refactor : LottoFactoryTest 리팩터링

hongsii commented on src/main/java/lotto/domain/LottoFactory.java:
```diff
@@ -70,4 +70,15 @@ public static Lottos create(int count) {
+        for (String input : list) {
+            lottos.add(create(input));
+        }
+        for(int i =0; i< autoCount;i++) {
```
띄어쓰기 컨벤션 위반입니다.

hongsii commented on src/main/java/lotto/domain/LottoCount.java:
```diff
@@ -0,0 +1,37 @@
+
+    public int getAutoLottoCount() {
+        return autoLotto;
+    }
```
해당 메서드는 자동/수동의 개수를 반환하는 getter인데 서로 네이밍 형식이 다르네요!

hongsii commented on src/main/java/lotto/domain/LottoFactory.java:
```diff
@@ -70,4 +70,15 @@ public static Lottos create(int count) {
         return new Lottos(lottos);
     }
+
+    public static Lottos create(List<String> list, int autoCount){
```
`list` 라는 이름만 보고 어떤 파라미터인지 유추하기 힘들지 않을까요?

hongsii commented on src/main/java/lotto/util/StringUtil.java:
```diff
@@ -15,4 +19,29 @@ public static String removeBlank(String value) {
+	}
+
+	public static void checkRange(String input) {
+		if(Integer.parseInt(input) < 0)
```
여기만 괄호가 없는 이유가 무엇일까요? 일관된 포맷을 가져가면 좋을 것 같습니다~

hongsii commented on src/main/java/lotto/view/InputView.java:
```diff
@@ -19,4 +21,18 @@ public static String getBonusNumber() {
         return scanner.nextLine();
     }
+
+    public static String getCount() {
```
`getCount()` 는 모호한 네이밍입니다. 적절한 네이밍으로 바꿔보면 어떨까요?

hongsii commented on src/main/java/lotto/domain/LottoFactory.java:
```diff
@@ -70,4 +70,15 @@ public static Lottos create(int count) {
+        }
+        for(int i =0; i< autoCount;i++) {
+            lottos.add(create());
+        }
```
메서드를 추출해 가독성을 높여보면 어떨까요?

hongsii commented on src/main/java/lotto/view/InputView.java:
```diff
@@ -19,4 +21,18 @@ public static String getBonusNumber() {
+    public static List<String> getManualLottosNumber(int manualLottoCount) {
+        List<String> input = new ArrayList<>();
+        System.out.println("수동으로 구매할 번호를 입력해 주세요.");
+        for(int i=0;i<manualLottoCount;i++){
```
컨벤션 위반입니다.

hongsii reviewed (changes requested):
2단계도 잘 구현해주셨습니다.
몇 가지 피드백드렸으니 확인하셔서 반영해주세요~

fucct committed:
refactor : 피드백 반영

컨벤션에 맞게 띄어쓰기, 중괄호 수정
메서드, 변수명 명확하게 수정
LottoFactory 내의 메소드 makeManualLottos 추출, addAutoLotto 추출

fucct committed:
refactor : 피드백 반영

컨벤션에 맞게 띄어쓰기, 중괄호 수정
메서드, 변수명 명확하게 수정
LottoFactory 내의 메소드 makeManualLottos 추출, addAutoLotto 추출

hongsii commented on src/main/java/lotto/domain/result/Count.java:
```diff
@@ -0,0 +1,13 @@
+package lotto.domain.result;
+
+public class Count {
```
이 클래스는 왜 추가되었을까요..?

hongsii commented on src/main/java/lotto/domain/LottoCount.java:
```diff
@@ -0,0 +1,37 @@
+        StringUtil.checkRange(input);
+    }
+
+    private void validateMoney(String input, int totalLotto) {
```
`input` 을 더 의미있는 이름으로 바꿔보면 좋을 듯 합니다.

hongsii reviewed (changes requested):
피드백 반영을 잘해주셨네요.
사소한 피드백 드렸으니, 여기까지 반영해주세요~

fucct committed:
refactor : 2/29 피드백 반영

input parameter 명 ManualLotto로 수정

hongsii commented on src/main/java/lotto/domain/result/Rank.java:
```diff
@@ -47,8 +47,8 @@ public boolean checkBonus(boolean isBonus) {
-    public boolean isBonus(){
-        if(this==BONUS){
+    public boolean isBonus() {
+        if (this == BONUS) {
```
여기는 아래와 같이 줄여쓸 수 있겠네요~

``` java
return this == BONUS
```

fucct replied:
안녕하세요! 피드백 잘 받았습니다! 저번에도 DM보내드렸는데 Count클래스 가 1단계때 함께 제출이 안된것 같아요! push 과정에서 제가 실수한 것 같아요 ㅠㅠ 보내주신 피드백은 수정했는데 Count 파일은 1단계 피드백 반영에 필요한 클래스라 뒤늦게 추가했습니다. 아마도 기존에 Count 클래스가 없어서 갑자기 추가되어서 삭제해달라고 피드백 주신 것 같은데 원래 포함되는게 맞는 클래스입니다!! 혼란스럽게 해서 죄송합니다 ㅠㅠ

hongsii replied:
음.. 1단계때 사용되었다는 말씀이신거죠~? `Count` 클래스를 사용하는 로직을 못 본 것 같아서 요청드린건데 포함되는게 맞다면 삭제하지 않으셔도 됩니다!

hongsii reviewed (changes requested):
피드백 반영 잘해주셨네요!
간단한 피드백 남겼으니, 이전에 피드백 드린 불필요한 파일 삭제와 함께 반영해주세요 :)

fucct committed:
refactor : 3/1 피드백 반영

Rank 의 isBonus메서드 리턴 방식 수정

hongsii reviewed (approved):
디디 미션진행하시느라 고생많으셨습니다!
머지할게요 :)
